pkg/kubelet/prober/prober.go

```diff
@@ -231,11 +231,7 @@ func (p *prober) newExecInContainer(container api.Container, containerID kubecon
 		var buffer bytes.Buffer
 		output := ioutils.WriteCloserWrapper(&buffer)
 		err := p.runner.ExecInContainer(containerID, cmd, nil, output, output, false, nil)
-		if err != nil {
-			return nil, err
-		}
-
-		return buffer.Bytes(), nil
+		return buffer.Bytes(), err
```

Comment:
In fact, it's kind of strange to consider an exit code != 0 as an error. The command was running. I don't consider that as an error for the `Cmd.Run`. But it's the API. Doesn't help to not like it ;)